src/MultivariateStats.jl

```diff
@@ -352,4 +355,165 @@ MLJBase.is_pure_julia(::Type{<:ICA}) = true
+relative to the within-class variance and classify based on bayes rule. 
+"""
+const LDAFields = """
+    ## Keywords
+    `method` selects the method to be used. \n
+    `shrinkage`
+    `out_dim`
+    `regcoef`
+    """
+
+"""
+` LDA( ; method=:gevd, shrinkage=:None , out_dim=1 , regcoef=1e-4 ) `
+
+$LDA_Desc
+
+$LDAFields
+
+See also the [package documentation](https://multivariatestatsjl.readthedocs.io/en/latest/lda.html)
+"""
+mutable struct LDA <: MLJBase.Probabilistic
+    method::Symbol
+    shrinkage::Union{Symbol,Float64}
+    out_dim::Int64
+    regcoef::Float64
+
+end
+
+function MLJBase.clean!(model::LDA)
+    warning = ""
+    if model.method ∉ [:gevd, :whiten, :subspace_lda]
+        warning *= "Unknown method. Resetting method=:gevd.\n"
+        model.method = :gevd
+    end
+    if model.method == :subspace_lda && model.shrinkage != :None
+        warning *= "shrinkage not implemented for :subspace_lda. Resetting to shrinkage=:None.\n"
+        model.shrinkage = :None
+    end
+    if typeof(model.shrinkage) <: Real && !(0 <= model.shrinkage <= 1)
+        warning *= "shrinkage has to be ∈ [0,1]. Resetting to shrinkage=:lw. \n"
+    elseif model.shrinkage ∉ [ :lw, :None ]
+        warning *= "Unknown model solver. Resetting to shrinkage=:None.\n"
+        model.shrinkage = :None
+    end
+    if model.out_dim < 1
+        warning *= "Need out_dim ≥ 1 and <=min(no_of_features,no_of_classes). Resetting out_dim=1.\n"
+        model.lambda = 1
+    end
+    if model.regcoef <= 0
+        warning *= "Need regcoef > 0 . Resetting regcoef=1e-4.\n"
+        model.regcoef = 1e-4
+    end
+    return warning
+end
+
+
+function LDA(; method=:gevd, shrinkage=:None, out_dim=1, regcoef=1e-6)
+    model = LDA(method, shrinkage, out_dim, regcoef)
+    message = MLJBase.clean!(model)
+    isempty(message) || @warn message
+    return model
+end
+
+function MLJBase.fit(model::LDA, verbosity::Int, X, y)
+ 
+    class_list = MLJBase.classes(y[1]) #class list containing unique entries in y
+    
+    # NOTE: copy/transpose
+    Xmatrix ::Array{Float64,2} = MLJBase.matrix(X,transpose=true)
+    training_sample_size ::Int64 = size(Xmatrix,2)
+
+    #convert the target array of categoricals (y) to array of integers (yarray)
+    #and get the number of classes from yarray
+    yarray = MLJBase.int(y)
+    no_of_classes = maximum(yarray)
+
+    #check to make sure output dimension isn't too large or negative
+    0 < model.out_dim <= min(no_of_classes-1, training_sample_size) ||
+     throw(ArgumentError("out_dim must lie within 0 and min(no_of_features,training_sample_size)"))
+
+    if model.method ∈ [:gevd, :whiten]
+        model.shrinkage == :None ?
+            covestimator ::Union{MS.SimpleCovariance,LinearShrinkage}=MS.SimpleCovariance() :
+            covestimator = LinearShrinkage(target=DiagonalCommonVariance(),shrinkage=model.shrinkage)
+
+        core_fitresult=MS.fit(MS.MulticlassLDA, Int(no_of_classes), Xmatrix, Int.(yarray);
```

Comment:
Are these `Int` necessary?

Reply:
yes, The function  as defined in MultivariateStats package requires integers of type Int while `MLJBase.int` function gives output of type `UInt`.

Reply:
right, I had in mind that `UInt <: Int` but that's not the case. thanks